File: Data Structures & Algorithms/find-duplicate-integer/submission-2.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        for i, n in enumerate(nums):
            if (i+1 == n):
                continue
            elif n < i + 1 or nums[n-1] == n:
                return n
            hold = nums[n-1]
            while(hold!=n):
                next_hold = nums[hold-1]
                if(hold == nums[hold-1]):
                    return hold
                nums[hold-1] = hold
                hold = next_hold
            nums[n-1] = n
        return 0

            
        

File: Data Structures & Algorithms/remove-element/submission-2.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int numValid = 0;
        for (int i=0; i<nums.size(); i++){
            if (nums[i] != val){
                nums[numValid] = nums[i];
                numValid++;
            }
        }
        return numValid;
    }
};